src/main/java/org/folio/rest/impl/UserImportAPI.java

```diff
@@ -473,168 +413,107 @@ private ImportResponse processSuccessfulImportResponse(List<Future<SingleUserImp
-                if (futurePermissionHandler.failed()) {
-                  LOGGER.error(FAILED_TO_REGISTER_PERMISSIONS + user.getExternalSystemId());
-                }
-                future.complete(SingleUserImportResponse.created(user.getExternalSystemId()));
-              });
-            } catch (Exception e) {
-              LOGGER.warn(FAILED_TO_REGISTER_PERMISSIONS + user.getExternalSystemId());
-              future.complete(getFailedUserResponse(user, e));
-            }
-          } else {
-            errorManagement(userCreationResponse, ex, future,
-              FAILED_TO_CREATE_NEW_USER_WITH_EXTERNAL_SYSTEM_ID + user.getExternalSystemId(),
-              SingleUserImportResponse.failed(user.getExternalSystemId(), user.getUsername(), userCreationResponse.getCode(),
-                FAILED_TO_CREATE_NEW_USER_WITH_EXTERNAL_SYSTEM_ID + user.getExternalSystemId()));
-          }
-        });
-    } catch (Exception exc) {
-      LOGGER.error(FAILED_TO_CREATE_NEW_USER_WITH_EXTERNAL_SYSTEM_ID + user.getExternalSystemId(), exc.getMessage());
-      future.complete(getFailedUserResponse(user, exc));
-    }
-    return future.future();
+    // create permission before user object to avoid user without permission
+    // a dangling permission user object does not hurt.
+    return addEmptyPermissionSetForUser(okapiHeaders, user).compose(x ->
+        HttpClientUtil.getRequestOkapi(HttpMethod.POST, okapiHeaders, userCreationQuery)
+            .expect(ResponsePredicate.SC_CREATED)
+            .sendJsonObject(JsonObject.mapFrom(user))
+            .compose(
+                res -> Future.succeededFuture(SingleUserImportResponse.created(user.getExternalSystemId())),
+                e -> Future.succeededFuture(SingleUserImportResponse.failed(user.getExternalSystemId(), user.getUsername(),
+                    500, FAILED_TO_CREATE_NEW_USER_WITH_EXTERNAL_SYSTEM_ID + user.getExternalSystemId()))));
+    // e.getMessage() not used
   }
 
   private Future<RequestPreference> createUserPreference(User user, UserImportData userImportData,
-                                                         Map<String, String> okapiHeaders) {
+      Map<String, String> okapiHeaders) {
 
     RequestPreference requestPreference = userImportData.getRequestPreferences().get(user.getUsername());
     if (Objects.nonNull(requestPreference)) {
       requestPreference.setUserId(user.getId());
       return prefService.validate(requestPreference, userImportData, user)
-        .compose(o -> {
-          udpService.updateUserPreference(requestPreference, userImportData);
-          return prefService.create(okapiHeaders, requestPreference);
-        });
+          .compose(o -> {
+            udpService.updateUserPreference(requestPreference, userImportData);
+            return prefService.create(okapiHeaders, requestPreference);
+          });
     } else {
-      return succeededFuture().mapEmpty();
+      return Future.succeededFuture().mapEmpty();
```

Comment:
`Future.succeededFuture().mapEmpty()` can be simplified to `Future.succeededFuture()`.
Easier to read guard clause:
```
if (requestPreference == null) {
  return succeededFuture();
}
requestPreference.setUserId(user.getId());
return ...
```